Lib/asyncio/streams.py

```diff
@@ -241,7 +242,7 @@ def connection_made(self, transport):
             res = self._client_connected_cb(reader,
                                             self._stream_writer)
             if coroutines.iscoroutine(res):
-                self._loop.create_task(res)
+                self._task = self._loop.create_task(res)
```

Comment:
This approach will not work for multiple connections at the same time.

```suggestion
                task = self._loop.create_task(res)
                task.add_done_callback(self._connection_tasks.remove)
                self._connection_tasks.add(task)
```